.asf.yaml

```diff
@@ -52,3 +52,24 @@ github:
     squash: true
     merge: false
     rebase: false
+
+  protected_branches:
+    master:
+      required_status_checks:
+        # strict means "Require branches to be up to date before merging".
+        strict: false
+        # contexts are the names of checks that must pass
+        contexts:
+          - Docker/build
+          - Frontend/build
+          - PR Lint/check
+          - E2E/Cypress
+          - Python Misc/lint
+          - Python MySQL/test-mysql
+          - Python MySQL/test-postgres
+          - Python MySQL/test-sqlite
```

Comment:
@nytai @ktmud Do we feel comfortable with this list or do we want to use the api response instead?

Reply:
I found a few other projects using this feature
https://github.com/apache/skywalking-eyes/blob/main/.asf.yaml
https://github.com/apache/skywalking/blob/master/.asf.yaml
https://github.com/apache/skywalking-python/blob/master/.asf.yaml

It seems they're using the portion after the `/`, which would result in duplicates for us. I think this looks good to me

Reply:
I suspect if a match isn't found in the list of recent contexts nothing happens

Reply:
I suspect it will wait for that check forever... Maybe let's add a very short list to override the admin settings first, then gradually add them back?

Reply:
When you go to settings you can't just enter any string, you have to choose from the list of "status checks found in the last week for this repository". I think it needs to be a context the repo has seen before. Also, here they have `build` set but no check is marked as required.
https://github.com/apache/skywalking-client-js/blob/04dd6b3b024f838b3b92002a370ce3b7f70638b4/.asf.yaml#L38

Reply:
That being said, it may be a good idea to be extra safe here as we may end up in a stuck state waiting for apache infra.

Reply:
Tested with a personal fork, now I'm 90% sure GitHub will just be forever waiting for these unidentifiable checks:

<img width="954" alt="API request" src="https://user-images.githubusercontent.com/335541/105833261-ca07ed80-5f7d-11eb-875d-cd1d6af9b9f4.png">

<img width="975" alt="settings" src="https://user-images.githubusercontent.com/335541/105833271-cecca180-5f7d-11eb-97af-2eb9e2b6b2f9.png">

<img width="963" alt="random-checks" src="https://user-images.githubusercontent.com/335541/105833304-d9873680-5f7d-11eb-964d-a717a46f4e9f.png">


IMO the safest course of actions is:

1. Update this PR to include only the simplest check (e.g. "PR Lint") in the required list, just to override manual admin settings.
2. Once the CI passes, merge it.
3. Make another PR to rename the jobs with ambiguous names
4. Maybe another PR to add other required checks back to the "required" list. Could combine with 3, too.

Reply:
ah, this is great. @ktmud did it looks like three of four of those context strings didn't match any existing checks. Did "build" work and match to `Frontend / build`?

Reply:
Actually based on your other screenshot, I think I should go with the whole string with the space `PR Lint / check` for example.

Reply:
Thanks for the thorough investigation @ktmud. I think you're right, this file is just used to make a request to the github api.